app/src/main/java/org/sopt/diablo/adapter/FollowerAdapter.kt

```diff
@@ -0,0 +1,48 @@
+package org.sopt.diablo.adapter
+
+import android.view.LayoutInflater
+import android.view.ViewGroup
+import androidx.recyclerview.widget.RecyclerView
+import org.sopt.diablo.data.UserData
+import org.sopt.diablo.databinding.ItemFollowerListBinding
+
+class FollowerAdapter: RecyclerView.Adapter<FollowerAdapter.FollowerViewHolder>() {
+    val userList = mutableListOf<UserData>()
+    private lateinit var itemClickListener: ItemClickListener
+
+    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): FollowerViewHolder {
+        val binding =
+            ItemFollowerListBinding.inflate(LayoutInflater.from(parent.context), parent, false)
+        return FollowerViewHolder(binding, itemClickListener)
+    }
+
+    override fun onBindViewHolder(holder: FollowerViewHolder, position: Int) {
+        holder.onBind(userList[position])
+    }
+
+    override fun getItemCount(): Int = userList.size
+
+    class FollowerViewHolder(
+        private val binding: ItemFollowerListBinding,
+        private val itemClickListener: ItemClickListener
+    ) : RecyclerView.ViewHolder(binding.root) {
+        fun onBind(data: UserData) {
+            binding.apply {
+                ivProfile.setImageResource(data.profile)
+                tvName.text = data.name
+                tvIntroduction.text = data.introduction
+            }
+            itemView.setOnClickListener {
+                itemClickListener.onClick(data)
+            }
+        }
+    }
+
+    interface ItemClickListener {
```

Comment:
```suggestion
    fun interface ItemClickListener {
```
코틀린에 추가된 문법으로 `interface` 앞에 `fun` 을 붙여 SAM(Single Abstract Method) 를 지원하는 인터페이스로 만들 수 있습니다. 이렇게 되면 인터페이스의 익명 객체를 만들 때 바로 람다로 사용이 되는 것으로 압니다.

```kotlin
    fun interface A {
        fun hi()
    }
    fun B(a: A) {}
    B { } // No problem
```

Reply:
그리고 이렇게 인터페이스를 만드는 과정 없이 생성자를 통해 람다로 전달해도 괜찮을 것 같습니다.

저 같은 경우 대략 다음과 같이 전달을 했습니다. 꼭 이게 좋은 방법이란것은 절대 아닙니다.
```kotlin
class GithubProfileAdapter(
    private val dataFlow: StateFlow<List<GithubProfile>>,
    private val menuOpenStateFlow: StateFlow<List<Boolean>>,
    private val lifecycle: Lifecycle,
    private val pixelRatio: PixelRatio,
    private val onItemMoveCallback: (Int, Int) -> Unit,
    private val onItemRemoved: (Int) -> Unit,
    private val onItemMenuOpened: (Int) -> Unit,
    private val onItemMenuClosed: (Int) -> Unit,
    private val onItemClicked: (GithubProfile, View) -> Unit,
) : RecyclerView.Adapter<GithubProfileAdapter.GithubFollowerHolder>(), MoveableAdapter {
```